serializer/serializer.h

```diff
@@ -34,6 +34,20 @@ extern "C" {
 #define DEFAULT_MSG_LEN 58
 /** @} */
 
+/**
+ * @brief Patch input string with nines into assigned length. It can be used for
+ * patching message or tag
+ *
+ * @param[out] new_str Output patched string
+ * @param[in] old_str Input string which needs to be patched
+ * @param[in] new_str_len assigned output string length
+ *
+ * @return
+ * - SC_OK on success
+ * - non-zero on error
+ */
+status_t fill_nines(char* new_str, char* old_str, size_t new_str_len);
```

Comment:
I have no idea why you would like to declare `fill_nines` as a public function.

Reply:
The query string of `api_find_transactions_by_tag()` doesn't need to be deserialized, since the query string is the plain `tag` we going to use in the next steps. If I didn't declare `fill_nines()` as a public function, then I need to declare a `find_transactions_by_tag_req_deserialize()` especially for `fill_nines()`. I thought it would we akward doing in thi way, so I decided to delcare it in public.

Reply:
`fill_nines` is not really generic from the perspective of API design. It was tended to behave as cross-functionality among internal components. We can move it to something named after `include/private` for internal headers.